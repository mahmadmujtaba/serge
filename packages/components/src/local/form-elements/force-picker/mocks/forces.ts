const forces = [
  {
    name: 'Blue Force',
    colour: '#69c',
    selected: true
  },
  {
    name: 'Red Force',
    colour: '#f00',
    selected: false
  },
  {
    name: 'White Force',
    colour: '#fff',
    selected: false
  }
]

export default forces
