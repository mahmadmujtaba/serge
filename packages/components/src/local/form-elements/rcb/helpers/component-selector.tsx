import React from 'react'

import { Checkbox, Radio } from '@material-ui/core'
import ComponentSelector from '../types/component-selector'

const componentSelector: ComponentSelector = (type, option, value, postBack) => {

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    postBack(event.target.value);
  };

  let output = null
  switch (type) {
    case 'checkbox' :
      output = <Checkbox name={option.toString().toLowerCase()} checked={option === value} onChange={handleChange} />
      break
    case 'radio' :
      output = <Radio />
      break
  }
  return output
}

export default componentSelector
