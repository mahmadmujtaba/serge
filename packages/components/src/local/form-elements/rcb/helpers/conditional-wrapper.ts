import CW from '../types/conditional-wrapper'

const ConditionalWrapper: CW = ({ condition, wrapper, children }): any =>
  condition ? wrapper(children) : children

export default ConditionalWrapper
