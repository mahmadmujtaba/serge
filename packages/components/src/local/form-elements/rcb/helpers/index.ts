/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as componentSelector } from './component-selector';
export { default as ConditionalWrapper } from './conditional-wrapper';