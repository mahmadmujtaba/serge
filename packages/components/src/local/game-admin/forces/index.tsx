import React, { useState } from 'react'

import cx from 'classnames'
/* Import proptypes */
import PropTypes, { ForceData, Role } from './types/props'

/* Import Styles */
import styles from './styles.module.scss'

/* Import Components */
import { Content, LeftSide, RightSide } from '../content'
import EditableList, { Item } from '../editable-list'
import TransparentInput from '../../form-elements/transparent-input'
import IconUploader from '../../molecules/icon-uploader'
import Button from '../../form-elements/button'
import Colorpicker from '../colorpicker'
import FormGroup from '../../form-elements/form-group-shadow'
import SortableList, { Item as SortableListItem } from '../../form-elements/sortable-list'
import PasswordView from '../password-view'
import Switch from '@material-ui/core/Switch'
import { withStyles } from '@material-ui/core/styles'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faEye, faComments, faDirections } from '@fortawesome/free-solid-svg-icons'
import { UMPIRE_FORCE } from '@serge/config'

const MobileSwitch = withStyles({
  switchBase: {
    color: '#FFFFFF',
    '&$checked': {
      color: '#1A394D'
    },
    '&$checked + $track': {
      backgroundColor: '#1A394D'
    }
  },
  checked: {},
  track: {}
})(Switch)

/* Render component */
export const Forces: React.FC<PropTypes> = ({ forces, onChange, onSave }) => {
  const [selectedItem, setSelectedItem] = useState(0)
  const handleSwitch = (_item: Item, key: number): void => {
    setSelectedItem(key)
  }

  const handleChangeForces = (nextForces: Array<ForceData>): void => {
    onChange(nextForces)
  }

  const renderContent = (): React.ReactNode => {
    if (!forces[selectedItem]) return null
    const data = forces[selectedItem]

    const handleChangeForce = (force: ForceData): void => {
      const nextForces: Array<ForceData> = [...forces]
      nextForces[selectedItem] = force
      handleChangeForces(nextForces)
    }

    const handleCreateRole = (): void => {
      const roles: Array<Role> = [...data.roles, {
        name: 'New Role',
        canSubmitPlans: false,
        password: 'p' + Math.random().toString(36).substring(8),
        control: false,
        isInsightViewer: false,
        isObserver: false
      }]
      handleChangeForce({ ...data, roles: roles })
    }

    const renderRoleFields = (item: SortableListItem, key: number): React.ReactNode => {
      const roleItem = item as Role
      const handleChangeRole = (nextRole: Role, submitPlans = false): void => {
        const roles: Array<Role> = submitPlans ? data.roles.map(role => ({ ...role, canSubmitPlans: false })) : [...data.roles]
        roles[key] = nextRole
        handleChangeForce({ ...data, roles })
      }

      return (
        <div className={styles.role}>
          <div className={styles['role-item']}>
            <PasswordView value={roleItem.password} onChange={(password: string): void => {
              handleChangeRole({ ...roleItem, password })
            }}/>
            {key === 0 && <div className={styles['role-title']}>Password</div>}
          </div>
          <div className={styles['role-item']}>
            <MobileSwitch disabled={data.uniqid !== UMPIRE_FORCE} size='small' checked={roleItem.isObserver} onChange={(): void => {
              handleChangeRole({ ...roleItem, isObserver: !roleItem.isObserver })
            }} />
            {key === 0 && <div
              title='Can view all channels'
              className={cx(styles['role-title'], styles['title-center'])}>
              <FontAwesomeIcon icon={faEye} />
            </div>}
          </div>
          <div className={styles['role-item']}>
            <MobileSwitch disabled={data.uniqid !== UMPIRE_FORCE} size='small' checked={roleItem.isInsightViewer} onChange={(): void => {
              handleChangeRole({ ...roleItem, isInsightViewer: !roleItem.isInsightViewer })
            }} />
            {key === 0 && <div
              title='Can view feedback/insights'
              className={cx(styles['role-title'], styles['title-center'])}>
              <FontAwesomeIcon icon={faComments} />
            </div>}
          </div>
          <div className={styles['role-item']}>
            <MobileSwitch size='small' checked={roleItem.canSubmitPlans} onChange={(): void => {
              handleChangeRole({ ...roleItem, canSubmitPlans: !roleItem.canSubmitPlans }, !roleItem.canSubmitPlans)
            }} />
            {key === 0 && <div
              title='Can submit mapping plans'
              className={cx(styles['role-title'], styles['title-center'])}>
              <FontAwesomeIcon icon={faDirections} />
            </div>}
          </div>
        </div>
      )
    }

    return (
      <div key={selectedItem}>
        <div className={styles.row}>
          <div className={styles.col}>
            <TransparentInput
              onChange={(name: string): void => {
                handleChangeForce({ ...data, name })
              }}
              value={data.name}
              placeholder="Platform Name"/>
          </div>
          <div className={styles['color-box']}>
            <Colorpicker value={data.color} onChange={(newColor: string): void => {
              handleChangeForce({ ...data, color: newColor })
            }} />
          </div>
          <div className={styles.col}>
            <IconUploader limit={20000} icon={data.icon} onChange={(icon: string): void => {
              handleChangeForce({ ...data, icon })
            }}>Change Icon</IconUploader>
          </div>
          <div className={styles.actions}>
            <Button onClick={(): void => { if (onSave) onSave() }}>Save</Button>
          </div>
        </div>
        <div className={styles.row}>
          <div className={cx(styles.col, styles.section)}>
            <FormGroup placeholder="Overview & Objectives">
              <textarea rows={8} className={styles.textarea} onChange={(e): void => {
                handleChangeForce({ ...data, overview: e.target.value })
              }} value={data.overview} />
            </FormGroup>
          </div>
        </div>
        <div className={styles.row}>
          <div className={cx(styles.col, styles.section)}>
            <FormGroup placeholder="Roles">
              <SortableList
                remove={false}
                sortable='auto'
                required
                onChange={(roles: Array<SortableListItem>): void => {
                  handleChangeForce({ ...data, roles: roles as Array<Role> })
                }}
                onCreate={handleCreateRole}
                renderItemSection={renderRoleFields}
                items={data.roles}
                title='Add Role' />
            </FormGroup>
          </div>
        </div>
      </div>
    )
  }

  return (
    <Content>
      <LeftSide>
        <EditableList items={forces} onClick={handleSwitch} />
      </LeftSide>
      <RightSide>
        {renderContent()}
      </RightSide>
    </Content>
  )
}

export default Forces

export { ForceData } from './types/props'
