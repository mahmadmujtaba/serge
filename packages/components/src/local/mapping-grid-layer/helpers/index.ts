import toHex from './to-hex'
import hexesNamed from './hex-named'
import hexesBetween from './hexes-between'
import hexesInRange from './hexes-in-range'
import cellForLocation from './cell-for-location'
import cellIndexToDegrees from './cell-index-to-degrees'
import cellIndexToRelativeDegrees from './cell-index-to-relative-degrees'

export {
  toHex,
  cellForLocation,
  cellIndexToDegrees,
  cellIndexToRelativeDegrees,
  hexesNamed,
  hexesBetween,
  hexesInRange
}
