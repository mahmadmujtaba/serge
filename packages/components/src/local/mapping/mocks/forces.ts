export const forces = [
    {
      color: '#FCFBEE',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'White',
      overview: 'Umpire force.',
      roles: [
        {
          control: true,
          isInsightViewer: true,
          isObserver: true,
          name: 'Game Control',
          password: 'p2311'
        }
      ],
      umpire: true,
      uniqid: 'umpire'
    },
    {
      assets: [
        {
          name: 'HMS DEVONSHIRE',
          contactId: 'C043',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'frigate',
          position: 'S23',
          status: {
            state: 'Transiting',
            speedKts: 20
          },
          uniqid: 'a0pra00001'
        },
        {
          name: 'STING',
          contactId: 'C072',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'helicopter',
          position: 'P01',
          status: {
            state: 'Landed'
          },
          uniqid: 'a0pra00002'
        },
        {
          name: 'P8',
          contactId: 'C076',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'fixed-wing-aircraft',
          position: 'P03',
          status: {
            state: 'Landed'
          },
          uniqid: 'a0pra00012'
        },
        {
          name: 'EDDYSTONE PT',
          contactId: 'C012',
          condition: 'Full capability',
          perceptions: {
            Red: {
              force: 'Blue',
              type: ''
            }
          },
          platformType: 'merchant-vessel',
          position: 'S23',
          status: {
            state: 'Transiting',
            speedKts: 20
          },
          uniqid: 'a0pra00003'
        }
      ],
      color: '#00F',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'Blue',
      overview: 'Blue force.',
      roles: [
        {
          control: false,
          isInsightViewer: false,
          isObserver: false,
          name: 'CO',
          password: 'p5543'
        }
      ],
      umpire: false,
      uniqid: 'Blue'
    },
    {
      assets: [
        {
          name: 'TARRADA',
          contactId: 'C065',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'fishing-vessel',
          position: 'N07',
          status: {
            state: 'Transiting',
            speedKts: 10
          },
          uniqid: 'a0pra000100'
        },
        {
          name: 'WBIED-1',
          contactId: 'C108',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'torpedo',
          position: 'T01',
          status: {
            state: 'Onboard'
          },
          uniqid: 'a0pra020102'
        },
        {
          name: 'WBIED-2',
          contactId: 'C118',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'torpedo',
          position: 'T03',
          status: {
            state: 'Onboard'
          },
          uniqid: 'a0pra130102'
        },
        {
          name: 'Osa',
          contactId: 'C003',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {
            Blue: {
              force: 'Red',
              type: ''
            }
          },
          platformType: 'fast-attack-craft',
          position: 'M05',
          status: {
            state: 'Moored'
          },
          uniqid: 'a0pra160102'
        },
        {
          name: 'Yevgenya',
          contactId: 'C313',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {
            Blue: {
              force: 'Red',
              type: 'fast-attack-craft'
            }
          },
          platformType: 'mcmv',
          position: 'M04',
          status: {
            state: 'Moored'
          },
          uniqid: 'a0pra170102'
        },
        {
          name: 'Coastal Radar C',
          contactId: 'C476',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'coastal-radar-site',
          position: 'Q12',
          status: {
            state: 'Inactive'
          },
          uniqid: 'a0prc000103'
        }
      ],
      color: '#F00',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'Red',
      overview: 'Red force.',
      roles: [
        {
          control: false,
          isInsightViewer: false,
          isObserver: false,
          name: 'CO',
          password: 'p3244'
        }
      ],
      umpire: false,
      uniqid: 'Red'
    }
  ]


export default forces
