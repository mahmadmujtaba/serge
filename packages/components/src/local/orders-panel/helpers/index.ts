/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as panelTitle } from './panel-title'
export { default as showOrders } from './show-orders'
