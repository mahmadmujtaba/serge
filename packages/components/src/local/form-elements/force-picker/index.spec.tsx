/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import ForcePicker from './index'

import forces from './mocks/forces'

import { getSelectedItem } from '@serge/helpers'

it('CheckboxType renders correctly', () => {
  const tree = renderer
    .create(<ForcePicker label="Visible to" options={forces} selected={getSelectedItem(forces).name} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
