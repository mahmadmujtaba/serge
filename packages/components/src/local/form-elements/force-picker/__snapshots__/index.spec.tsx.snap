// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['CheckboxType renders correctly 1'] = `
<section
  className="input-container"
>
  <h2>
    Visible to
  </h2>
  <div
    aria-label="Visible to"
    className="MuiFormGroup-root"
    role="radiogroup"
  >
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-17"
            name="visible to"
            onChange={[Function]}
            type="radio"
            value="Blue Force"
          />
          <div
            className="PrivateRadioButtonIcon-root-30 PrivateRadioButtonIcon-checked-32"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-31"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        <div>
          <span
            className="color-box"
            style={
              Object {
                "backgroundColor": "#69c",
              }
            }
          />
          Blue Force
        </div>
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-17"
            name="visible to"
            onChange={[Function]}
            type="radio"
            value="Red Force"
          />
          <div
            className="PrivateRadioButtonIcon-root-30"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-31"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        <div>
          <span
            className="color-box"
            style={
              Object {
                "backgroundColor": "#f00",
              }
            }
          />
          Red Force
        </div>
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-17"
            name="visible to"
            onChange={[Function]}
            type="radio"
            value="White Force"
          />
          <div
            className="PrivateRadioButtonIcon-root-30"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-31"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        <div>
          <span
            className="color-box"
            style={
              Object {
                "backgroundColor": "#fff",
              }
            }
          />
          White Force
        </div>
      </span>
    </label>
  </div>
</section>
`
